src/installer/pkg/sfx/installers/osx_scripts/host/postinstall

```diff
@@ -10,9 +10,21 @@ INSTALL_DESTINATION=$2
 # A temporary fix for the permissions issue(s)
 chmod 755 $INSTALL_DESTINATION/dotnet
 
-# Add the installation directory to the system-wide paths
-# But first create the directory if it doesn't exist
-mkdir -p /etc/paths.d
-echo $INSTALL_DESTINATION | tee /etc/paths.d/dotnet
+if [ -e /etc/dotnet/install_location ]; then
+    # clear out any entries for this architecture if they exist
+    sed -i old '/^${InstallerTargetArchitecture}=/d' /etc/dotnet/install_location
+else
+    mkdir -p /etc/dotnet    
+fi
+
+echo ${InstallerTargetArchitecture}=$INSTALL_DESTINATION | tee -a /etc/dotnet/install_location
+
+# if we're running on the native architecture
+if [[ "$(uname -m)" =~ "${UnameMachineRegex}" ]]; then
```

Comment:
`uname -m` will give you x86_64 under rosetta emulation. Is this the intended behavior?

Reply:
If someone is really running this installer package under Rosetta emulation, then yes I think that should be the expected behavior.

Reply:
I added a card to our board to test this.  I'm not so worried about accidentally setting path, that's easy to fix and user can change it per their intent.  I am worried if somehow we can lie to the installer javascript detection, since that would put the files  in the wrong place.  @sfoslund

Reply:
> If someone is really running this installer package under Rosetta emulation, then yes I think that should be the expected behavior.

I believe this is against what was proposed - only setting path of dotnet for the native one. I'm ok with it either way.

Reply:
> I believe this is against what was proposed  - only setting path of dotnet for the native one.

If user’s intention is to “Run as x86_64 app” on arm64 host, in that case “native” is the target of emulation/virtualization and x86_64 behavior is expected. This is how other software (including Apple’s own) is handling this case.

Reply:
Do either of you have any pointer for how one might force a .pkg to install under emulation?  Is that a thing?  Perhaps trying to launch from a rosetta terminal (would it flow that way?)

Reply:
Interesting, I just tried this out and I think I found a way to test it.

If I create a terminal that runs using rosetta, then launch the installer from that ternminal I see it report `hw.machine` as `x86_64` on my M1.  This is not good, we don't want the installer to be lied to.

@sfoslund here's the command I ran to do this:
`sudo installer -dumplog -verboseR -pkg testPkg.pkg -target /`

We should check the newer detection you added and see if that's avoiding this redirection.

Reply:
I just tried the repro with latest detection and it also sees the x64 values when run from rosetta terminal.

I found at least one way to identify when this is happening: https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment.

`sysctl.proc_translated`  --> this returns 1 when run from rosetta.  I wonder if that's a sufficient escape hatch for our purposes.

I opened https://github.com/dotnet/runtime/issues/59370 to track this.

Reply:
> This is not good, we don't want the installer to be lied to.

I was considering it as a correct behavior based on how other installers (node.js, rustup and so on) operate in Rosetta2, and more generally, any emulation environments (qemu). Forcing our will against user wishes is no good, IMO.

Reply:
Based on my quick research, it looks like we can detect if we're running in rosetta with `sysctl -in sysctl.proc_translated`. Do we just want to try to block the install in this case? Or should we add this check to our logic for determining if a machine is x64?

Reply:
@am11 perhaps you can open an issue with a scenario that you feel is blocked and we can discuss that?

We're actually making the runtime aware of emulation and making it so both x64 and ARM64 can coexist.  The problem is that existing installers weren't aware and would clash.  We're preserving the single entrypoint, so we do have to make the installers aware of real architecture so that they can avoid clashing.  This is the part we can't have lied to, since that circumvents the work.  The rest of the runtime is happy to be lied to, that's what the support for this is all about.